src/web/api/v1/git.rs

```diff
@@ -91,20 +91,121 @@ async fn get_commit(
 
 fn get_commit_by_prefix<'repo>(
     repo: &'repo git2::Repository,
-    commit_id: &str,
+    rev: &str,
 ) -> Result<git2::Commit<'repo>, api::AppError> {
-    repo.find_commit_by_prefix(commit_id)
-        .map_err(|e| match e.code() {
-            git2::ErrorCode::Invalid => {
-                api::AppError::BadRequest(format!("Invalid base commit ID '{commit_id}': {e}"))
+    let oid = get_object_for_revision(repo, rev)?.id();
+    repo.find_commit(oid).map_err(|e| match e.code() {
+        git2::ErrorCode::Invalid => {
+            api::AppError::BadRequest(format!("Invalid base commit ID '{rev}': {e}"))
+        }
+        git2::ErrorCode::NotFound => {
+            api::AppError::NotFound(format!("Base commit '{rev}' not found: {e}"))
+        }
+        _ => api::AppError::InternalServerError(format!("Failed to find base commit '{rev}': {e}")),
+    })
+}
+
+#[derive(Serialize, ToSchema, PartialEq)]
+#[serde(rename_all = "lowercase")]
+enum DiffType {
+    Binary,
+    Text,
+}
+
+#[derive(Serialize, ToSchema)]
+#[serde(rename_all = "camelCase")]
+struct DiffFile {
+    /// Path to the diff file
+    path: Option<String>,
+    /// Content of the diff file
+    content: Option<String>,
+}
+
+impl DiffFile {
+    pub fn try_from_repo_and_diff_file(
+        repo: &git2::Repository,
+        diff_file: &git2::DiffFile,
+    ) -> Option<DiffFile> {
+        let oid = diff_file.id();
+        if oid.is_zero() {
+            log::warn!(
+                "OID is zero '{:?}'",
+                diff_file
+                    .path()
```

Comment:
nit: below (l. 146) we also do not use these mappings to log the path, maybe skip it here. Or align the usage with a helper function to get the path as a string ( we need it anyhow for the return value)